platform/mbed_retarget.cpp

```diff
@@ -120,6 +121,17 @@ static void init_serial() {
 #endif
 }
 
+/**
+ * Sets errno when file opening fails.
+ * Wipes out the filehandle too.
+ */
+static int handle_open_errors(int error, unsigned filehandle_idx) {
+    errno = -error;
```

Comment:
> Certainly should be a comment that the error number passed in is negative.  

@geky if you can describe it somewhere along the code - will help.  we did not find it in the commit or code was not that clear during review.